src/svgs/runin/runin-shmetro.tsx

```diff
@@ -235,17 +329,153 @@ const NextText = (props: { nextName: Name } & React.SVGProps<SVGGElement>) => {
         </g>
     );
 };
+const StationNumberText = (props: { lineNum: string, stationNum: string } & React.SVGProps<SVGGElement>) => {
+    const { lineNum, stationNum, ...others } = props;
+    return (
+        <g textAnchor="middle" {...others}>
+            <text className="rmg-station-name" fontSize={22}
+                dy={0}>
+                {lineNum}
+            </text>
+            <text className="rmg-station-name" fontSize={22}
+                dy={28}>
+                {stationNum}
+            </text>
+        </g>
+    );
+};
+
+const StationNumberTextNow = (props: { lineNum: string, stationNum: string } & React.SVGProps<SVGGElement>) => {
+    const { lineNum, stationNum, ...others } = props;
+    return (
+        <g textAnchor="middle" {...others}>
+            <text className="rmg-station-name" fontSize={29}
+                dy={0}>
+                {lineNum}
+            </text>
+            <text className="rmg-station-name" fontSize={29}
+                dy={35}>
+                {stationNum}
+            </text>
+        </g>
+    );
+};
+
+
+const StationNumberNext = (props: { xpos: number, ypos: number, lineName: string, stationNumber: string } & React.SVGProps<SVGGElement>) => {
+    const { xpos, ypos, lineName, stationNumber, ...others } = props;
+    return (
+        <g >
+            {useMemo(
+                () => (
+                    <>
+                        <circle cx={xpos} cy={ypos} r="35" stroke="var(--rmg-theme-colour)" strokeWidth="8" fill="white">
+                        </circle>
+                        <line x1={xpos - 30} y1={ypos} x2={xpos + 30} y2={ypos} stroke="rgb(0,0,0)" strokeWidth="2" />
+                        <StationNumberText transform={"translate(" + (xpos).toString() + "," + (ypos - 5).toString() + ")"} lineNum={lineName} stationNum={stationNumber} />
+                    </>
+                ),
+                // eslint-disable-next-line react-hooks/exhaustive-deps
+                undefined
```

Comment:
`useMemo` can not be fed with `undefined`, learn it before use or throw memo away.